### PR TITLE
Milestone 4/5

### PR DESCRIPTION
**Pull request by ekarlson238**
Both milestone 4 and 5
Game is done and working correctly

### CODE REVIEW

**Review by hhoare — APPROVED** [review]
So many files, but all the scripts look good and follow the style guide.

**Review by RSand18 — APPROVED** [review]
Definitely a TON of files being changed/removed, which is something I would take note of for the future, but in terms of the actual code, I hadn't noticed anything significant.

File: Assets/Scripts/InventoryManager.cs [hunk 1]
```diff
@@ -4,17 +4,90 @@
 
 public class InventoryManager : MonoBehaviour
```
Comment (dantogno) [inline Assets/Scripts/InventoryManager.cs:5]: What's the intent of this class? Should all these fields be replaced with a list?

File: Assets/Scripts/InventoryObject.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            SceneValueManager.item3PickedUp = true;
+        if (this.objectName == "Campfire")
+            SceneValueManager.item4PickedUp = true;
+        #endregion
+
+        if (this.objectName == "Key")
+            SceneValueManager.keyPickedUp = true;
+
+    }
+
```
Comment (dantogno) [inline Assets/Scripts/InventoryObject.cs:89]: remove extra lines

File: Assets/Scripts/Portal.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        base.InteractWith();
+        SceneValueManager.playerEnteredPortal = true;
+        StartCoroutine(FadeOut());
+    }
+
+    IEnumerator FadeOut()
+    {
+        fadePanelAnim.SetBool("leavingScene", true);
+        yield return new WaitForSeconds(1f);
```
Comment (dantogno) [inline Assets/Scripts/Portal.cs:26]: why 1 second? Magic number.

File: Assets/Scripts/SceneValueManager.cs [hunk 4]
```diff
@@ -10,33 +10,24 @@ public class SceneValueManager : MonoBehaviour
     /// Example: The player picks up an item off the ground in a scene; after leaving and returning to the scene the item doesnt reappear
     /// </summary>
 
+    public static bool playerEnteredPortal = false;
+
     #region key bools
     public static bool keyPickedUp = false;
```
Comment (dantogno) [inline Assets/Scripts/SceneValueManager.cs:16]: avoid public fields, use properties (even for static)

File: Assets/Scripts/Wander.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+public class Wander : MonoBehaviour
+{
+    [SerializeField]
+    private float wanderRadius;
+
+    [Tooltip("How long before a new destination is randomized?")]
+    [SerializeField]
+    private float wanderTimer;
+
+    //private Transform target;
```
Comment (dantogno) [inline Assets/Scripts/Wander.cs:15]: remove commented out code

File: Assets/Scripts/Wander.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class Wander : MonoBehaviour
+{
+    [SerializeField]
+    private float wanderRadius;
+
+    [Tooltip("How long before a new destination is randomized?")]
```
Comment (dantogno) [inline Assets/Scripts/Wander.cs:11]: good tooltip, name was not enough for me to "get it"

File: Assets/Scripts/Wander.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+        if (timer >= wanderTimer)
+        {
+            Vector3 newPos = RandomNavSphere(transform.position, wanderRadius, -1);
+            agent.SetDestination(newPos);
+            timer = 0;
+        }
+    }
+
+    public static Vector3 RandomNavSphere(Vector3 origin, float dist, int layermask)
```
Comment (dantogno) [inline Assets/Scripts/Wander.cs:37]: you should add a summary for this, especially because it's public and presumably called from other classes.

Comment (dantogno) [reply Assets/Scripts/Wander.cs:37]: And if it's not necessary to call from outside classes, make it private!

**Pull request merged**